lib/Exception/DocumentReferencedException.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager\Exception;
+
+use PHPCR\PropertyInterface;
+
+/**
+ * This is exception is thrown if a document is still referenced, and the executed action does not work because of that.
+ */
+class DocumentReferencedException extends DocumentManagerException
+{
+    /**
+     * @var object
+     */
+    private $document;
+
+    /**
+     * @var PropertyInterface[]
+     */
+    private $references;
+
+    /**
+     * @param object $document
+     * @param PropertyInterface[] $references
+     * @param \Exception $previous
+     */
+    public function __construct($document, array $references = [], \Exception $previous = null)
+    {
+        $this->document = $document;
+        $this->references = $references;
+
+        parent::__construct(
+            sprintf(
+                'The document with the object hash "%s" cannot be deleted because it is still referenced',
```

Comment:
Would prefer this to be an inline message, as with all the other exception messages in this libary:

````php
throw new BlahException(sprintf('This is the message'));
````

As we have discussed before. It is clearer to read in the source code, and usually each instance that an exception is thrown has a slightly different context - usually information can be given to help the developer solve their problems.

If the exception is only used in one place, then putting the message in the Exception class just means that information (i.e. explanation for why exception would be thrown) that would be available when reading the source has to be looked up in a new class.

I would prefer, regardless of other policies, that we maintain the current standard in this library.